protocol/xmrtaker/swap_state.go

```diff
@@ -154,13 +123,100 @@ func newSwapState(
 		exchangeRate,
 		ethAsset,
 		stage,
-		walletScanHeight,
+		moneroStartNumber,
 		statusCh,
 	)
 	if err = b.SwapManager().AddSwap(info); err != nil {
 		return nil, err
 	}
 
+	defer func() {
+		statusCh <- stage
+	}()
```

Comment:
Do we want to execute this even if `newSwapState` below returns an error?

Reply:
updated so it executes only after all err checks